src/peer_manager.rs

```diff
@@ -1141,6 +1158,48 @@ impl PeerManager {
             .collect()
     }
 
+    /// Returns the `PeerId`s and names of all the peers held in the RT, and whether they are
+    /// connected via a tunnel or not.
+    pub fn get_routing_peer_details(&self) -> Vec<(PeerId, XorName, bool)> {
+        self.routing_table
+            .iter()
+            .filter_map(|name| -> Option<(PeerId, XorName, bool)> {
+                let peer = match self.peer_map.get_by_name(name) {
+                    Some(peer) => peer,
+                    None => {
+                        error!("{:?} has {} in RT, but has no entry in peer_map for it.",
+                               self,
+                               name);
+                        return None;
+                    }
+                };
+                let peer_id = match peer.peer_id {
+                    Some(peer_id) => peer_id,
+                    None => {
+                        error!("{:?} has {} in RT, but has no peer ID for it.", self, name);
+                        return None;
+                    }
+                };
+                let is_tunnel = match peer.state {
+                    PeerState::Routing(RoutingConnection::Tunnel) => true,
+                    PeerState::Routing(_) => false,
+                    _ => {
+                        error!("{:?} has {} in RT, but has state {:?} for it.",
+                               self,
+                               name,
+                               peer.state);
+                        return None;
+                    }
+                };
+                Some((peer_id, *name, is_tunnel))
+            })
+            .collect()
+    }
+
+    pub fn correct_routing_state_to_direct(&mut self, peer_id: &PeerId) {
+        let _ = self.set_state(peer_id, PeerState::Routing(RoutingConnection::Direct));
```

Comment:
It's not the comments I care about here, it's the fact that if the function's return type is changed, ignoring it might no longer be the correct thing to do. Using a type guard helps there.